File: code/community/1301690756020310020/implementing-language-detection-deepgram-api.cs
using System;
using System.Net.Http;
using System.Threading.Tasks;

class Program
{
    private static readonly HttpClient client = new HttpClient();

    static async Task Main()
    {
        var apiKey = Environment.GetEnvironmentVariable("DEEPGRAM_API_KEY");

        client.DefaultRequestHeaders.Add("Authorization", $"Token {apiKey}");

        var url = "https://api.deepgram.com/v1/listen?detect_language=true";

        var audioBytes = await File.ReadAllBytesAsync("your_audio_file.mp3");
        var audioContent = new ByteArrayContent(audioBytes);
        audioContent.Headers.ContentType = new System.Net.Http.Headers.MediaTypeHeaderValue("audio/mpeg");

        var response = await client.PostAsync(url, audioContent);
        var responseString = await response.Content.ReadAsStringAsync();

        Console.WriteLine(responseString);
    }
}

File: code/community/1301690756020310020/implementing-language-detection-deepgram-api.go
package main

import (
	"bytes"
	"fmt"
	"io/ioutil"
	"net/http"
	"os"
)

func main() {
	apiKey := os.Getenv("DEEPGRAM_API_KEY")

	audioData, err := ioutil.ReadFile("your_audio_file.mp3")
	if err != nil {
		panic(err)
	}

	url := "https://api.deepgram.com/v1/listen?detect_language=true"

	req, err := http.NewRequest("POST", url, bytes.NewReader(audioData))
	if err != nil {
		panic(err)
	}

	req.Header.Set("Authorization", fmt.Sprintf("Token %s", apiKey))
	req.Header.Set("Content-Type", "audio/mpeg")

	client := &http.Client{}
	resp, err := client.Do(req)
	if err != nil {
		panic(err)
	}
	defer resp.Body.Close()

	body, err := ioutil.ReadAll(resp.Body)
	if err != nil {
		panic(err)
	}

	fmt.Println(string(body))
}